### ISSUE TITLE
Spectrum syncs from 0% on every start

### ISSUE THREAD

**Issue by jjim1982** [cryptoadvance/specter-desktop#2364]
I'm not sure if this is expected behaviour or not, but on every start spectrum needs to sync from 0% (spinning icon in top right bitcoin connection).

MyNode 0.3.18, electrum server fully synced
Specter 2.0.1 (MacOS)

**Comment by csjui** [comment]
Any comment on this? Has anyone got spectrum/specter working properly with MacOS?

**Comment by Funman2** [comment]
my spectrum node connection on Start9 embassy is always showing sync status 0% , never changes, but tx are all correctly listed, haven't tried a fresh tx since high fees, but will test, seems like a bug.

**Comment by k9ert** [comment]
Yes, this is expected behaviour. You won't realise until you (re-) start your specter which is usually not the case for specter running in node-implementations.
However, this should be quick unless you have dozens of wallets. How many wallets do you have and how long takes the sync?

**Comment by csjui** [comment]
> Yes, this is expected behaviour. You won't realise until you (re-) start your specter which is usually not the case for specter running in node-implementations. However, this should be quick unless you have dozens of wallets. How many wallets do you have and how long takes the sync?

When I restart myNode for example, electrum server doesn't need to resync from scratch though? I guess spectrum doesn't have a similar persistent db structure then?

**Comment by Funman2** [comment]
> Yes, this is expected behaviour. You won't realise until you (re-) start your specter which is usually not the case for specter running in node-implementations. However, this should be quick unless you have dozens of wallets. How many wallets do you have and how long takes the sync?

Yes, actually I have a wallet with 100+ tx and 8 more wallets. I actually gave up running Specter on this rasperry pi node with electrum since loading times where beyond my patience limit, it took way longer then using Bitcoin core connection. I am running the Start9 implementation.
But with Bitcoin core connection get this infamous key error (can't load UTXOs) with the big wallet,  which is a bug ever since version 2.0, I does not have this error in version 1.14.4, so I basically have up on specter in node implementations now. 

Actually I have to mention with umbrel I have a fast server , not a pi, there I also had this key error in new specter version,  but after a while the wallet loaded fine.But on rasperry pi it never loads.
But on umbrel I don't have https connection,  so I can't scan with my Seedsigner,  using Sparrow and nunchuk now...

**Comment by k9ert** [comment]
Hopefully at some point i get to some performance improvements.
